### PR TITLE
Rework predictor builders

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

**Comment by regadas** [comment]
@andrewsmartin @ravwojdyla PTAL

File: zoltar-core/src/main/java/com/spotify/zoltar/DefaultPredictorBuilder.java [hunk 1]
```diff
@@ -128,32 +128,14 @@
 
   public abstract AsyncPredictFn<ModelT, InputT, VectorT, ValueT> predictFn();
 
-  public abstract Builder<ModelT, InputT, VectorT, ValueT> toBuilder();
+  public abstract Predictor<InputT, ValueT> predictor();
 
   @Override
-  public Predictor<InputT, ValueT> predictor(
```
Comment (regadas) [inline zoltar-core/src/main/java/com/spotify/zoltar/DefaultPredictorBuilder.java]: This is a method that shouldn't be exposed. Hence this PR

File: zoltar-core/src/main/java/com/spotify/zoltar/DefaultPredictorBuilder.java [hunk 2]
```diff
@@ -128,32 +128,14 @@
...
+  public DefaultPredictorBuilder<ModelT, InputT, VectorT, ValueT> with(
```
Comment (andrewsmartin) [inline zoltar-core/src/main/java/com/spotify/zoltar/DefaultPredictorBuilder.java]: What's the point of this method now? Maybe I'm missing something but isn't this just a duplicate of `create()`?

Comment (andrewsmartin) [reply zoltar-core/src/main/java/com/spotify/zoltar/DefaultPredictorBuilder.java]: Like this is an instance method but just ends up returning a new one via `create`

**Pull request merged**